libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_serializer.c

```diff
@@ -114,45 +114,37 @@ TEST_GROUP_RUNNER( Provisioning_Unit_Serializer )
  *"{}" empty
  * payload.
  */
-TEST( Provisioning_Unit_Serializer, TestCreateKeysAndCertificateSerializationNominalCase )
+TEST( Provisioning_Unit_Serializer, TestSerializeCreateKeysAndCertificatePayloadNominalCase )
 {
-    IotSerializerEncoderObject_t testEncoder = IOT_SERIALIZER_ENCODER_CONTAINER_INITIALIZER_STREAM;
-
     /* The request payload is an empty map container that needs. */
     const uint8_t pExpectedSerialization[] =
     {
         0xA0 /*# map( 0 ) */
     };
 
-    uint8_t pSerializationBuffer[ sizeof( pExpectedSerialization ) ] = { 0 };
-
-    /* Test the serializer function without a valid buffer. */
-    TEST_ASSERT_EQUAL( AWS_IOT_PROVISIONING_SUCCESS,
-                       _AwsIotProvisioning_SerializeCreateKeysAndCertificateRequestPayload( &testEncoder,
-                                                                                            NULL,
-                                                                                            0 ) );
-    TEST_ASSERT_EQUAL( sizeof( pExpectedSerialization ),
-                       _pAwsIotProvisioningEncoder->getExtraBufferSizeNeeded( &testEncoder ) );
-    _pAwsIotProvisioningEncoder->destroy( &testEncoder );
-
+    uint8_t * pSerializationBuffer = NULL;
+    size_t bufferSize = 0;
 
-    /* Test the serializer function without a valid buffer. */
+    /* Test the serializer function. */
     TEST_ASSERT_EQUAL( AWS_IOT_PROVISIONING_SUCCESS,
-                       _AwsIotProvisioning_SerializeCreateKeysAndCertificateRequestPayload( &testEncoder,
-                                                                                            pSerializationBuffer,
-                                                                                            sizeof( pSerializationBuffer ) ) );
-
-    _pAwsIotProvisioningEncoder->destroy( &testEncoder );
+                       _AwsIotProvisioning_SerializeCreateKeysAndCertificateRequestPayload( &pSerializationBuffer,
+                                                                                            &bufferSize ) );
+    TEST_ASSERT_NOT_NULL( pSerializationBuffer );
```

Comment:
When will this be NULL in a test like this? Seems like you should mock Malloc and force it to fail to get this to fail, which would be a good test, but at this point the test should always pass?

Reply:
This is testing that the serializer function under test is allocating memory for the buffer pointer that was passed from the test to it as an output parameter. The initialized value of the buffer pointer in the test  is `NULL`.
I'm not sure what your suggestion of making `malloc` fail with the mock is intended for...

Reply:
Is it to test that the serializer function will return an error code if the memory allocation fails?

Reply:
Just a general comment about testing. You should answer that question, if we do not know why we added this assert then I would be worried. I was just pointing out that it seems like this will never assert under test conditions, maybe I am wrong :)